server/avl/seqset.go

```diff
@@ -250,9 +276,19 @@ var (
 	ErrSetNotEmpty = errors.New("ss: set not empty")
 )
 
-func Decode(buf []byte) (*SequenceSet, error) {
-	if len(buf) < minLen || buf[0] != magic || buf[1] != version {
-		return nil, ErrBadEncoding
+// Decode returns the sequence set and number of bytes read from the buffer on success.
+func Decode(buf []byte) (*SequenceSet, int, error) {
+	if len(buf) < minLen || buf[0] != magic {
+		return nil, -1, ErrBadEncoding
+	}
+
+	// For v1 version with 64 buckets per node.
+	if buf[1] == 1 {
+		return decodev1(buf)
+	}
+
+	if buf[1] != version {
```

Comment:
Would it be worth splitting out the below logic to `decodev2` and change this to a switch statement?

Reply:
Sure..